core/translate/optimizer.rs

```diff
@@ -39,12 +41,96 @@ pub fn optimize_plan(mut select_plan: Plan) -> Result<(Plan, ExpressionResultCac
         &select_plan.referenced_tables,
         &select_plan.available_indexes,
     )?;
+    eliminate_unnecessary_orderby(
+        &mut select_plan.root_operator,
+        &select_plan.referenced_tables,
+        &select_plan.available_indexes,
+    )?;
     find_shared_expressions_in_child_operators_and_mark_them_so_that_the_parent_operator_doesnt_recompute_them(&select_plan.root_operator, &mut expr_result_cache);
     Ok((select_plan, expr_result_cache))
 }
 
+fn _operator_is_already_ordered_by(
+    operator: &mut Operator,
+    key: &mut ast::Expr,
+    available_indexes: &Vec<Rc<Index>>,
+) -> Result<bool> {
+    match operator {
+        Operator::Scan {
+            table,
+            table_identifier,
+            ..
+        } => {
+            let tuple = (table.clone(), table_identifier.clone());
```

Comment:
lazy shit here with the string cloning just to make it work w the existing method signature of the trait methods .is_primary_key_of()